FernFlower-Patches/0004-Fix-field-initalizers.patch

```diff
@@ -181,15 +181,15 @@ index bec95a5..94ec37f 100644
          case Exprent.EXPRENT_FIELD:
 -          return false;
 +          FieldExprent fexpr = (FieldExprent)expr;
-+          if (cl.qualifiedName.equals(fexpr.getClassname())) {
++          if (cl.hasField(fexpr.getName(), fexpr.getDescriptor().descriptorString)) {
```

Comment:
this should probably be something like hasField and fieldIndex < myIndex?

Reply:
The field index is checked below. I changed it to use `hasField` because `getClassname` may return the current class, but the field referenced actually belongs to a super class. In this case, the method would return `false` because the field is not in the whitelist. Instead, this case is now passed to the else branch where we check if the field in the super class is static and can be safely referenced.

I hope I explained that well enough.

Reply:
I did similar but different fix @ https://github.com/thiakil/ForgeFlower/commit/4a15cad6c67359d2b3db2d014596a1cecfb2ff0b